plugins/modules/helm.py

```
@@ -526,6 +537,15 @@ def deploy(
     if reuse_values is not None:
         deploy_command += " --reuse-values=" + str(reuse_values)
 
+    if reset_then_reuse_values:
+        helm_version = module.get_helm_version()
+        if LooseVersion(helm_version) < LooseVersion("3.14.0"):
+            module.warn(
```

it it ok to have working, or better to change to fail like:
```
            module.fail_json(
                msg='helm support option --reset-then-reuse-values starting release >= 3.14.0',
                **opt_result
```

just to avoid a case when helm will runs without `--reset-then-reuse-values` with unexpected result in CI

---

You are right, we should fail when trying to use that kind of unsupported feature, I'll update the PR accordingly.